subprojects/packagefiles/libgrapheme/meson.build

```diff
@@ -0,0 +1,78 @@
+project(
+  'libgrapheme',
+  'c',
+  version: '2.0.2',
+  default_options: [
+    'c_std=c99',
+    'warning_level=3',
+    'b_lundef=false',
+    'buildtype=minsize',
+  ],
+  meson_version: '>=0.57.0',
+)
+
+foreach native : [true, false]
+  add_project_arguments(
+    '-D_DEFAULT_SOURCE',
+    language: 'c',
+    native: native,
+  )
+endforeach
+
+pkg_mod = import('pkgconfig')
+cc = meson.get_compiler('c')
+cc_id = cc.get_id()
+os_name = target_machine.system()
+
+subdir('src')
+
+subdir('gen')
+
+version = meson.project_version()
+version_split = version.split('.')
+soversion = '@0@.@1@'.format(version_split[0], version_split[1])
+lib_c_args = []
+lib_link_args = []
+
+# To produce the same binary as with the Makefile approach, we need to
+# add the -ffreestanding and -nostdlib flags
+if get_option('b_lundef') == false and cc.has_argument('-ffreestanding') and host_machine.system() != 'darwin'
+
+  lib_c_args += ['-ffreestanding']
+  lib_link_args += ['-nostdlib']
```

Comment:
These are unusual arguments for a desktop build.  I don't fully understand the consequences, though GCC docs say that `-nostdlib` usually wants at least `-lgcc`.  Willing to defer to upstream here, just wondering if this could cause issues.